### PR TITLE
Upgrade eventing

### PR DESCRIPTION
**Pull request by radufa**
**Description**
Upgrade Kyma-Eventing

Changes proposed in this pull request:

- Extend the "upgrade_kyma" process for using the Service Manager
- Add new installing steps for Eventing-EMS
- Deprovision Eventing Event-Hub

**Related issue(s)**

See also https://github.com/kyma-project/kyma/issues/10317

### CODE REVIEW

File: components/kyma-environment-broker/internal/process/upgrade_kyma/ems_provision_test.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/process/provisioning"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/Peripli/service-manager-cli/pkg/types"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/logger"
+
+	//"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/process/provisioning"
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/process/upgrade_kyma/ems_provision_test.go:14]: minor: remove the commented line

Comment (radufa) [reply components/kyma-environment-broker/internal/process/upgrade_kyma/ems_provision_test.go:14]: Removed

File: components/kyma-environment-broker/internal/process/provisioning/ems_provision.go [hunk 2]
```diff
@@ -62,10 +62,29 @@ func (s *EmsProvisionStep) Run(operation internal.ProvisioningOperation, log log
 
 func (s *EmsProvisionStep) provision(smCli servicemanager.Client, operation internal.ProvisioningOperation, log logrus.FieldLogger) (internal.ProvisioningOperation, time.Duration, error) {
 
+	input := GetEventingProvisioningData(operation.Ems)
+	resp, err := smCli.Provision(operation.Ems.Instance.BrokerID, *input, true)
+	if err != nil {
+		return s.handleError(operation, err, log, fmt.Sprintf("Provision() call failed for brokerID: %s; input: %#v", operation.Ems.Instance.BrokerID, input))
+	}
+	log.Infof("response from EMS provisioning call: %#v", resp)
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/process/provisioning/ems_provision.go:70]: please change to debug

File: components/kyma-environment-broker/internal/process/upgrade_kyma/ems_provision.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	return operation, 0, nil
+}
+
+func (s *EmsUpgradeProvisionStep) provision(smCli servicemanager.Client, operation internal.UpgradeKymaOperation, log logrus.FieldLogger) (internal.UpgradeKymaOperation, time.Duration, error) {
+	input := provisioning.GetEventingProvisioningData(operation.Ems)
+	resp, err := smCli.Provision(operation.Ems.Instance.BrokerID, *input, true)
+	if err != nil {
+		return s.handleError(operation, err, log, fmt.Sprintf("Provision() call failed for brokerID: %s; input: %#v", operation.Ems.Instance.BrokerID, input))
+	}
+	log.Infof("response from EMS provisioning call: %#v", resp)
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/process/upgrade_kyma/ems_provision.go:68]: change to debug

File: components/kyma-environment-broker/internal/process/upgrade_kyma/event_hub_step.go [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+}
+
+func (s DeprovisionAzureEventHubStep) Run(operation internal.UpgradeKymaOperation, log logrus.FieldLogger) (
+	internal.UpgradeKymaOperation, time.Duration, error) {
+	if operation.EventHub.Deleted {
+		log.Info("Event Hub is already deprovisioned")
+		return operation, 0, nil
+	}
+	hypType := hyperscaler.Azure
+	log.Infof("HAP lookup for credentials to deprovision cluster for global account ID %s on Hyperscaler %s",
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/process/upgrade_kyma/event_hub_step.go:49]: debug

File: components/kyma-environment-broker/internal/process/upgrade_kyma/service_manager_offering_step.go [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+	plans, err := smCli.ListPlansByName(s.planName, offerings.ServiceOfferings[0].ID)
+	if err != nil {
+		return s.handleError(operation, err, "unable to get Service Manager plan", log)
+	}
+	if len(plans.ServicePlans) != 1 {
+		return s.operationManager.OperationFailed(operation,
+			fmt.Sprintf("expected one %s Service Manager plan, but found %d", s.offeringName, len(offerings.ServiceOfferings)))
+	}
+	info.PlanID = plans.ServicePlans[0].CatalogID
+	log.Infof("Found plan: catalogID=%s", info.PlanID)
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/process/upgrade_kyma/service_manager_offering_step.go:78]: change to debug, also: "Found plan: planID="

File: components/kyma-environment-broker/internal/process/provisioning/ems_provision.go (hunk 2 above)
Comment (radufa) [reply components/kyma-environment-broker/internal/process/provisioning/ems_provision.go:70]: done

File: components/kyma-environment-broker/internal/process/upgrade_kyma/ems_provision.go (hunk 3 above)
Comment (radufa) [reply components/kyma-environment-broker/internal/process/upgrade_kyma/ems_provision.go:68]: done

File: components/kyma-environment-broker/internal/process/upgrade_kyma/event_hub_step.go (hunk 4 above)
Comment (radufa) [reply components/kyma-environment-broker/internal/process/upgrade_kyma/event_hub_step.go:49]: done

File: components/kyma-environment-broker/internal/process/upgrade_kyma/service_manager_offering_step.go (hunk 5 above)
Comment (radufa) [reply components/kyma-environment-broker/internal/process/upgrade_kyma/service_manager_offering_step.go:78]: done

**Review by piotrmiskiewicz — APPROVED** [review]
lgtm!!!

**Pull request merged**